cli/pcluster/commands.py

```diff
@@ -635,18 +635,37 @@ def status(args):  # noqa: C901 FIXME!!!
         sys.exit(0)
 
 
-def delete(args):
-    saw_update = False
-    LOGGER.info("Deleting: %s", args.cluster_name)
-    stack_name = utils.get_stack_name(args.cluster_name)
+def _persist_cloudwatch_log_group(args):
+    """Set cluster's CloudWatch log group to persist past cluster deletion via its DeletionPolicy."""
+    LOGGER.info("Configuring {}'s CloudWatch log group to persist past cluster deletion.".format(args.cluster_name))
+
+    # Ensure CloudWatch logging is enabled. If not, the log group won't exist.
+    stack = utils.get_cloudwatch_logs_substack(args.cluster_name)
+    if not utils.get_cloudwatch_logs_cfn_params(stack.get("Parameters")).get("enable"):
+        utils.error("CloudWatch logging is not enabled for cluster {}".format(args.cluster_name))
+
+    # Get and modify the substack's template, then update the stack w/ the modified version
+    try:
+        template = utils.get_template_for_stack(stack.get("StackName"))
+        if template.get("Resources").get("CloudWatchLogGroup").get("DeletionPolicy") != "Retain":
+            template["Resources"]["CloudWatchLogGroup"]["DeletionPolicy"] = "Retain"
+            utils.update_stack_template(stack.get("StackName"), template, stack.get("Parameters"))
+    except SystemExit as sys_exit:
```

Comment:
why catching SystemExit?

Reply:
That's what the error function does in the utils module. I figured since there was no stack trace to be printed out, catching it and "extending" the error message would give more context as to what went wrong.